### PR TITLE
Enable ANSI colors for Windows 10 and newer

### PR DESCRIPTION
**Pull request by hacdias**
See issue #496.

### CODE REVIEW

**Comment by sirupsen** [comment]
Have you tested this on all relevant versions of Windows? We don't have CI for this, so things go a bit havoc if it's merged and doesn't work across platforms.

**Comment by hacdias** [comment]
It only works for Windows 10 and newer because this functionality was announce for Win10.

**Comment by hacdias** [comment]
Just had a lot of troubles squashing. It was my first time 😆 and I made some mistakes. Anyway, I added a version check to ensure that the feature is only enabled on Windows 10 and newer. Although I will try to check if running the code on older versions harms or not.

**Comment by sirupsen** [comment]
This is really the simplest way to check Windows version? 😬

**Comment by sirupsen** [comment]
@rnapier you wrote this, can you review? I have zero experience with Windows 😅

**Comment by hacdias** [comment]
@sirupsen I wish I could have found a better way. There is a syscall for that but it was deprecated on Windows 8.1 and now, unless you direct your app exclusively for Windows 10 on the manifest data, it will always say you are on Windows 8. I tried that. It would be like... 5 lines?

**Comment by sirupsen** [comment]
Ok, let's wait for @rnapier to give this a spin.

Have you tested this on a Windows machine on versions less than 10?

File: terminal_windows.go [hunk 1]
```diff
@@ -18,8 +23,51 @@ var kernel32 = syscall.NewLazyDLL("kernel32.dll")
...
+	cmd.Stderr = stderr
+	err := cmd.Run()
+	if err != nil {
+		return -1, err
+	}
+
+	version := strings.Replace(stdout.String(), "\n", "", -1)
+	version = strings.Replace(version, "\r\n", "", -1)
+
+	x1 := strings.Index(version, "[Version")
```
Comment (sirupsen) [inline terminal_windows.go]: Can you add a comment with the output?

File: terminal_windows.go [hunk 2]
```diff
@@ -18,8 +23,51 @@ var kernel32 = syscall.NewLazyDLL("kernel32.dll")
...
+		return -1, errors.New("Can't determine Windows version")
+	}
+
+	return strconv.ParseFloat(version[x1+9:x1+13], 64)
+}
+
+func init() {
+	ver, err := getVersion()
+	if err != nil {
+		panic(err)
```
Comment (sirupsen) [inline terminal_windows.go]: I don't think we need to panic if we can't find the version, let's just below not do anything.

File: terminal_windows.go (hunk 1 above)
Comment (hacdias) [inline terminal_windows.go]: Done 😄

File: terminal_windows.go (hunk 2 above)
Comment (hacdias) [inline terminal_windows.go]: Sure! 😄

**Comment by hacdias** [comment]
> Have you tested this on a Windows machine on versions less than 10?

I haven't, but since the code doesn't run if you're not on Windows 10 or newer, I don't think there is any problem.

**Comment by Kushcabbage** [comment]
I know this is a year old but, is there a way to enable ANSI color in a .bat script for a java console program?

**Pull request merged**